src/plugins/intel_gna/src/transformations/ts_concat.cpp

```diff
@@ -0,0 +1,129 @@
+
+    const Shape concat_output_shape = concat_node->get_output_shape(0);
+    const int64_t axis = concat_node->get_concatenation_axis();
+    if (get_num_first_one_dims(concat_output_shape) != axis)
+        return false;
+
+    for (size_t i = 0; i < concat_node->get_input_size(); ++i) {
+        auto concat_input = concat_node->input_value(i);
+        auto transpose = ov::as_type_ptr<Transpose>(concat_input.get_node_shared_ptr());
+        if (transpose && !Limitations::is_transpose_supported(transpose))
+            return true;
+    }
+
+    return false;
+}
+
+}  // namespace
+
+TSConcatForward::TSConcatForward() {
+    MATCHER_SCOPE(TSConcatForward);
+
+    auto concat_node_label = wrap_type<Concat>(is_sinked);
+
+    matcher_pass_callback matcher_pass_callback = [=](Matcher& m) {
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        auto& concat_node_output = pattern_to_output.at(concat_node_label);
+        auto concat_node = as_type_ptr<Concat>(concat_node_output.get_node_shared_ptr());
+
+        std::vector<AxisVector> gather_indices_vecs;
+        OutputVector concat_inputs;
+        for (size_t i = 0; i < concat_node->get_input_size(); ++i) {
+            auto concat_input = concat_node->input_value(i);
+            auto transpose = ov::as_type_ptr<Transpose>(concat_input.get_node_shared_ptr());
+            const bool is_transposed_input = transpose && !Limitations::is_transpose_supported(transpose);
+
+            auto input_node_output = concat_input;
+            if (is_transposed_input)
+                input_node_output = concat_input.get_node_shared_ptr()->input_value(0);
+
+            const Shape& input_shape = input_node_output.get_shape();
+            const size_t input_dims =
+                std::accumulate(input_shape.begin(), input_shape.end(), 1, std::multiplies<Shape::value_type>());
+            Shape reshape_shape = {1, input_dims};
+            auto reshape_input_const = std::make_shared<Constant>(ov::element::i64, ov::Shape{2}, reshape_shape);
+            auto reshape_input = std::make_shared<Reshape>(input_node_output, reshape_input_const, false);
+            concat_inputs.push_back(reshape_input->output(0));
+
+            ov::copy_runtime_info(concat_node, {reshape_input, reshape_input_const});
+
+            if (is_transposed_input) {
+                auto transpose_const = ov::as_type_ptr<Constant>(transpose->get_input_node_shared_ptr(1));
+                if (!transpose_const)
+                    return false;
+                auto gather_indices_value =
+                    make_gather_indices_from_transpose_axes(transpose->get_input_shape(0),
+                                                            transpose_const->get_axis_vector_val());
+                gather_indices_vecs.push_back(gather_indices_value);
+            } else {
+                const Shape& input_shape = concat_input.get_shape();
```

Comment:
it incorrectly generates indexes for inputs without Transpose, please take the implementation from layout transformations branch

Reply:
Discussed on the meeting. We will create separate PR for this transformation